### PR TITLE
Use preview metadata for sizing attachment messages

### PR DESCRIPTION
**Pull request by chromakode**
:eyeglasses: @keybase/react-hackers

### CODE REVIEW

File: shared/actions/chat.js [hunk 1]
```diff
@@ -675,8 +683,12 @@ function _unboxedToMessage (message: MessageUnboxed, idx: number, yourName, conv
           }
         case CommonMessageType.attachment:
           // $FlowIssue
-          const preview = payload.messageBody.attachment.preview
+          const preview: Assert = payload.messageBody.attachment.preview
```
Comment (chrisnojima) [inline shared/actions/chat.js]: Asset?

File: shared/actions/chat.js [hunk 2]
```diff
@@ -675,8 +683,12 @@ function _unboxedToMessage (message: MessageUnboxed, idx: number, yourName, conv
...
           const mimeType = preview && preview.mimeType
+          let previewSize
+          if (preview && preview.metadata.image) {
```
Comment (chrisnojima) [inline shared/actions/chat.js]: preview.metadata null check?

File: shared/actions/chat.js (hunk 1 above)
Comment (chromakode) [inline shared/actions/chat.js]: My bad -- pushed an unfinished commit up. Fixed. :sweat:

File: shared/actions/chat.js (hunk 2 above)
Comment (chromakode) [inline shared/actions/chat.js]: `preview.metadata` is not a maybe type in the new updated types

**Pull request merged**